styx-scheduler-service/src/main/java/com/spotify/styx/api/SchedulerResource.java

```diff
@@ -161,9 +162,15 @@ public SchedulerResource(
     }
 
     try {
-      stateManager.receive(event);
+      if ("dequeue".equals(EventUtil.name(event))) {
```

Comment:
How about only allowing `dequeue` here for backwards compatibility and rejecting all other event types?

Reply:
Done